src/main/java/de/tk/opensource/privacyproxy/util/ProxyHelper.java

```diff
@@ -17,12 +22,48 @@ public class ProxyHelper {
     public static final int ROUTING_TIMEOUT_MILLISECONDS = 5000;
     private static final Logger LOGGER = LoggerFactory.getLogger(ProxyHelper.class);
 
-    private final Proxy proxy;
+    private final String proxyHost;
+    private final Integer proxyPort;
     private final String nonProxyHosts;
+    private final HttpHost httpProxyHost;
 
-    public ProxyHelper(Proxy proxy, String nonProxyHosts) {
-        this.proxy = proxy;
+    private Proxy proxy;
+    private DefaultRoutePlanner proxyRoutePlanner;
+
+    public ProxyHelper(Proxy proxy, final String proxyHost, final Integer proxyPort, final String nonProxyHosts) {
+        this.proxyHost = proxyHost;
+        this.proxyPort = proxyPort;
         this.nonProxyHosts = nonProxyHosts;
+        this.proxy = proxy;
+        this.httpProxyHost = (this.proxyHost != null && this.proxyPort != null) ? new HttpHost(this.proxyHost, this.proxyPort) : null;
+    }
+
+    private Proxy getProxy() {
+        if (proxy == null) {
+            if (proxyHost == null || proxyPort == null) {
+                proxy = Proxy.NO_PROXY;
+            } else {
+                proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyHost, proxyPort));
+            }
+        }
+        return proxy;
+    }
```

Comment:
Was hältst du hier von early return + ternary?
```suggestion
    private Proxy getProxy() {
        if (proxy != null) {
            return proxy;
        }

        return (proxyHost == null || proxyPort == null) 
            ? Proxy.NO_PROXY 
            : new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyHost, proxyPort));
    }
```